central/group/datastore/datastore_test.go

```diff
@@ -699,16 +719,110 @@ func (s *groupDataStoreTestSuite) TestValidateGroup() {
 
-		err = s.dataStore.Mutate(s.hasWriteAccessCtx, nil, nil, []*storage.Group{g})
+		err = s.dataStore.Mutate(s.hasWriteAccessCtx, false, nil, nil, []*storage.Group{g})
 		s.Error(err)
 		s.ErrorIs(err, errox.InvalidArgs)
 	}
 }
+
+func (s *groupDataStoreTestSuite) TestUpdateMutableToImmutable() {
+	s.storage.EXPECT().Get(gomock.Any(), gomock.Any()).Return(&storage.Group{
+		Props: &storage.GroupProperties{
+			Id:             "id",
+			AuthProviderId: "apid",
+			Traits: &storage.Traits{
+				MutabilityMode: storage.MutabilityMode_ALLOW,
+			},
+		},
+	}, true, nil).Times(1)
+	s.storage.EXPECT().Upsert(gomock.Any(), gomock.Any()).Return(nil).Times(1)
+
+	err := s.dataStore.Update(s.hasWriteCtx, &storage.Group{
+		Props: &storage.GroupProperties{
+			Id:             "id",
+			AuthProviderId: "apid",
+			Traits: &storage.Traits{
+				MutabilityMode: storage.MutabilityMode_ALLOW_FORCED,
+			},
+			Key:   "abc",
+			Value: "dfg",
+		},
+		RoleName: "Admin",
+	})
+	s.NoError(err)
+}
+
+func (s *groupDataStoreTestSuite) TestUpdateImmutableError() {
+	s.storage.EXPECT().Get(gomock.Any(), gomock.Any()).Return(fixtureImmutableGroup(), true, nil).Times(1)
+
+	err := s.dataStore.Update(s.hasWriteCtx, &storage.Group{
+		Props: &storage.GroupProperties{
+			Id:             "id",
+			AuthProviderId: "apid",
+			Traits: &storage.Traits{
+				MutabilityMode: storage.MutabilityMode_ALLOW,
+			},
+		},
+		RoleName: "Admin",
+	})
+	s.ErrorIs(err, errox.InvalidArgs)
+}
+
+func (s *groupDataStoreTestSuite) TestDeleteImmutableNoForce() {
+	s.storage.EXPECT().Get(gomock.Any(), gomock.Any()).Return(fixtureImmutableGroup(), true, nil).Times(1)
+
+	err := s.dataStore.Remove(s.hasWriteCtx, false, &storage.GroupProperties{
+		Id:             "id",
+		AuthProviderId: "apid",
+	})
+	s.ErrorIs(err, errox.InvalidArgs)
+}
+
+func fixtureImmutableGroup() *storage.Group {
+	return &storage.Group{
+		Props: &storage.GroupProperties{
+			Id:             "id",
+			AuthProviderId: "apid",
+			Traits: &storage.Traits{
+				MutabilityMode: storage.MutabilityMode_ALLOW_FORCED,
+			},
+			Key:   "abc",
+			Value: "dfg",
+		},
+	}
+}
+
+func (s *groupDataStoreTestSuite) TestDeleteImmutableForce() {
+	s.storage.EXPECT().Get(gomock.Any(), gomock.Any()).Return(fixtureImmutableGroup(), true, nil).Times(1)
+	s.storage.EXPECT().Delete(gomock.Any(), gomock.Any()).Return(nil).Times(1)
+
+	err := s.dataStore.Remove(s.hasWriteCtx, true, &storage.GroupProperties{
+		Id:             "id",
+		AuthProviderId: "apid",
+	})
+	s.NoError(err)
+}
+
+func (s *groupDataStoreTestSuite) TestDefaultGroupCannotBeImmutable() {
+	group := &storage.Group{
+		Props: &storage.GroupProperties{
+			Id:             "id",
+			AuthProviderId: "apid",
+			Traits: &storage.Traits{
+				MutabilityMode: storage.MutabilityMode_ALLOW_FORCED,
+			},
+		},
+	}
+	err := s.dataStore.Update(s.hasWriteCtx, group)
+	s.ErrorIs(err, errox.InvalidArgs)
+	err = s.dataStore.Add(s.hasWriteCtx, group)
+	s.ErrorIs(err, errox.InvalidArgs)
+}
```

Comment:
For the mutable / immutable tests, could you also add some more tests for the `Mutate` call, to check that this also works as expected?